community/main.py

```diff
@@ -206,50 +205,225 @@ def update(self, delta_time):
-			self.f3.draw()
-
 		# CPU - GPU Sync
-		if not self.options.SMOOTH_FPS:
+		if not self.window.options.SMOOTH_FPS:
 			# self.fences.append(gl.glFenceSync(gl.GL_SYNC_GPU_COMMANDS_COMPLETE, 0))
 			# Broken in pyglet 2; glFenceSync is missing
 			pass
 		else:
 			gl.glFinish()
 
-	# input functions
-
 	def on_resize(self, width, height):
+		super().on_resize(width, height)
 		logging.info(f"Resize {width} * {height}")
 		gl.glViewport(0, 0, width, height)
 
 		self.player.view_width = width
 		self.player.view_height = height
-		self.f3.y = self.height - 10
-		self.f3.width = self.width // 3
+
+
+class MenuScene(Scene):
+	def __init__(self, window: Window) -> None:
+		super().__init__(window)
+		self.texture_manager = texture_manager.TextureManager(0, 0, 0)
+
+		self.icon = self.texture_manager.load_texture("dirt")
+		self.logo = self.texture_manager.load_texture("logo")
+
+	def on_draw(self):
+		super().on_draw()
+		gl.glEnable(gl.GL_TEXTURE_2D)
+		self.window.clear()
+
+	def update_ui(self):
+		super().update_ui()
+		io = imgui.get_io()
+		imgui.set_next_window_size(io.display_size.x, io.display_size.y)
+		imgui.set_next_window_position(0, 0)
+		imgui.push_style_var(imgui.STYLE_WINDOW_BORDERSIZE, 0.0)
+		imgui.push_style_var(imgui.STYLE_WINDOW_PADDING, (0.0, 0.0))
+
+		if imgui.begin(
+			"MC PY",
+			True,
+			flags=
+			imgui.WINDOW_NO_DECORATION |
+			imgui.WINDOW_ALWAYS_AUTO_RESIZE |
+			imgui.WINDOW_NO_SAVED_SETTINGS |
+			imgui.WINDOW_NO_NAV
+		):
+			# Get the draw list
+			draw_list = imgui.get_window_draw_list()
+
+			# Tile the image across the window
+			for x in range(0, int(io.display_size.x), self.icon[1]):
+				for y in range(0, int(io.display_size.y), self.icon[2]):
+					draw_list.add_image(self.icon[0].id, (x, y), (x + self.icon[1], y + self.icon[2]))
+
+			# Draw a semi-transparent overlay to darken the background
+			overlay_color = imgui.get_color_u32_rgba(0, 0, 0, 0.75)
+			draw_list.add_rect_filled(0, 0, io.display_size.x, io.display_size.y, overlay_color)
+
+
+			# Calculate the position to horizontally center the image
+			image_width = self.logo[1] / 2.5
+			image_height = self.logo[2] / 2.5
+
+			# Set the cursor position to the calculated center position
+			imgui.set_cursor_pos(((io.display_size.x - image_width) / 2, 50))
+
+			# Draw the image at the calculated position
+			imgui.image(self.logo[0].id, image_width, image_height, (0, 1), (1, 0))
+
+			# Calculate the position to horizontally center the buttons
+			button_width = 300  # Adjust button width as needed
+			button_height = 30  # Adjust button height as needed
+			button_x = (io.display_size.x - button_width) / 2
+			button_y = imgui.get_cursor_pos().y + image_height - 50
+
+			# Set the cursor position to the calculated center position for buttons
+			imgui.set_cursor_pos((button_x, button_y))
+
+			# Add buttons under the logo
+			if imgui.button("Singleplayer", width=button_width, height=button_height):
+				# Handle button 1 click
+				pass
```

Comment:
If I'm not mistaken, this means that there's currently no way to actually enter the world?

Reply:
That's correct for now, that's why this PR is "work in progress" / draft. I still need to implement the scene switching system.

Reply:
Resolved in [a97698f60d47c2b9ad4ef1f3ad2a60f702e044c8](https://github.com/obiwac/python-minecraft-clone/pull/104/commits/a97698f60d47c2b9ad4ef1f3ad2a60f702e044c8), the World class now takes a `save` parameter and the "Play tutorial level" button loads the exist world that we have. (There is no loading screen yet).